packages/salesforcedx-vscode-apex/src/views/testOutline.ts

```diff
@@ -0,0 +1,666 @@
+    this.passing = 0;
+    this.children.forEach(child => {
+      if ((child as ApexTest).passed) {
+        this.passing++;
+      }
+    });
+    this.label =
+      this.name + ' (' + this.passing + '/' + this.children.length + ')';
+    if (this.passing === this.children.length) {
+      this.updateIcon('Pass');
+    } else {
+      this.updateIcon('Fail');
+    }
+  }
+
+  public updateIcon(outcome: string) {
+    super.updateIcon(outcome);
+    if (outcome === 'Pass') {
+      this.children.forEach(child => {
+        // Update all the children as well
+        child.updateIcon(outcome);
+      });
+    }
+  }
+}
+
+export class ApexTest extends Test {
+  public errorMessage: string = '';
+  public stackTrace: string = '';
+  public passed = false;
+
+  constructor(
+    public label: string,
+    public file: vscode.Uri | null,
+    public row: number
+  ) {
+    super(label, vscode.TreeItemCollapsibleState.None, file, row);
+  }
+
+  public updateIcon(outcome: string) {
+    super.updateIcon(outcome);
+    if (outcome === 'Pass') {
+      this.errorMessage = '';
+      this.errorMessage = '';
+    }
+  }
+
+  public contextValue = 'apexTest';
+}
+
+class ReadableApexTestRunCodeActionExecutor extends (ForceApexTestRunCodeActionExecutor as {
+  new (test: string, shouldGetCodeCoverage: boolean): any;
+}) {
+  private outputToJson: string;
+  private apexTestOutline: ApexTestOutlineProvider;
+
+  public constructor(
+    tests: string[],
+    shouldGetCodeCoverage: boolean,
+    outputToJson: string,
+    apexTestOutline: ApexTestOutlineProvider
+  ) {
+    super(tests.join(','), shouldGetCodeCoverage);
+    this.outputToJson = outputToJson;
+    this.apexTestOutline = apexTestOutline;
+  }
+
+  public build(data: {}): Command {
+    this.builder = this.builder
+      .withDescription(
+        nls.localize('force_apex_test_run_codeAction_description_text')
+      )
+      .withArg('force:apex:test:run')
+      .withFlag('--tests', this.test)
+      .withFlag('--resultformat', 'human')
+      .withFlag('--outputdir', this.outputToJson)
+      .withFlag('--loglevel', 'error');
+
+    if (this.shouldGetCodeCoverage) {
+      this.builder = this.builder.withArg('--codecoverage');
+    }
+    return this.builder.build();
+  }
+
+  public async execute(response: ContinueResponse<{}>): Promise<void> {
+    const cancellationTokenSource = new vscode.CancellationTokenSource();
+    const cancellationToken = cancellationTokenSource.token;
+    const execution = new CliCommandExecutor(this.build(response.data), {
+      cwd: vscode.workspace.rootPath
+    }).execute(cancellationToken);
+
+    execution.processExitSubject.subscribe(async () => {
+      this.apexTestOutline.readJSONFile(this.outputToJson);
+    });
+
+    channelService.streamCommandOutput(execution);
+
+    if (this.showChannelOutput) {
+      channelService.showChannelOutput();
+    }
+
+    await ProgressNotification.show(execution, cancellationTokenSource);
```

Comment:
In a recent change, I removed the `await` on ProgressNotification.show and it should be removed here as well. After removing the `await`, the `async` should also be removed from line 640.